### PR TITLE
exp run/save: Add `--message`.

### PR DESCRIPTION
**Pull request by daavoo**
Per https://github.com/iterative/dvc/pull/9408

### CODE REVIEW

File: content/docs/command-reference/exp/run.md [hunk 1]
```diff
@@ -143,6 +144,10 @@ committing them to the Git repo. Unnecessary ones can be [cleared] with
 - `-C <path>`, `--copy-paths <path>` - List of ignored or untracked paths to
   copy into the temp directory. Only used if `--temp` or `--queue` is specified.
 
+- `-M <message>`, `--message <message>` - Custom commit message to use when
```
Comment (shcheklein) [inline content/docs/command-reference/exp/run.md:147]: is `-m` taken? (git has `-m`, so can be easier a bit)

File: content/docs/command-reference/exp/save.md [hunk 2]
```diff
@@ -63,6 +63,10 @@ committing them to the Git repo. Unnecessary ones can be [cleared] with
   directory to be included in the experiment. This option can be used multiple
   times.
 
+- `-M <message>`, `--message <message>` - Custom commit message to use when
```
Comment (shcheklein) [inline content/docs/command-reference/exp/save.md:66]: Custom -> custom

Comment (shcheklein) [reply content/docs/command-reference/exp/save.md:66]: `commiting the experiment` 

It reads to me like it's about `dvc exp apply` or `dvc exp branch` even. Is it about custom refs? Should we just use `save` or `create` an experiment here?

Comment (daavoo) [reply content/docs/command-reference/exp/save.md:66]: > Is it about custom refs? 

Not sure I follow the question, but yes, the actual git commit in the custom ref. 
When doing `exp branch` the branch will contain a commit with that message

File: content/docs/command-reference/exp/run.md (hunk 1 above)
Comment (shcheklein) [reply content/docs/command-reference/exp/run.md:147]: should we consider overriding this for `dvc exp run`? And get rid of it in 3.0? I don't even remember why did we have it. I think it was way before exp show, VS Code, Studio, etc.

File: content/docs/command-reference/exp/save.md (hunk 2 above)
Comment (shcheklein) [reply content/docs/command-reference/exp/save.md:66]: I mean, does add a message to a regular (non-commit) exp ref?

Comment (daavoo) [reply content/docs/command-reference/exp/save.md:66]: > I mean, does add a message to a regular (non-commit) exp ref?

Yes

Comment (shcheklein) [reply content/docs/command-reference/exp/save.md:66]: yep, so, I would probably use some neutral language them "saves an experiment" vs "commits".

File: content/docs/command-reference/exp/run.md (hunk 1 above)
Comment (daavoo) [reply content/docs/command-reference/exp/run.md:147]: Will do it in 3.0 (which should be soon) https://github.com/iterative/dvc/pull/9458

Comment (shcheklein) [reply content/docs/command-reference/exp/run.md]: does it means we'll have to rename it again then? (feels it's better to disable it on `dvc exp run` now tbh)

File: content/docs/command-reference/exp/save.md [hunk 3]
```diff
@@ -63,6 +63,9 @@ committing them to the Git repo. Unnecessary ones can be [cleared] with
   directory to be included in the experiment. This option can be used multiple
   times.
 
+- `-M <message>`, `--message <message>` - Custom message to use when saving the
```
Comment (shcheklein) [inline content/docs/command-reference/exp/save.md:66]: Custom -> custom

File: content/docs/command-reference/exp/run.md (hunk 1 above)
Comment (dberenbaum) [reply content/docs/command-reference/exp/run.md]: Why not merge with only `--message` and add `-m` when ready to break? That way, you don't need to wait for 3.0 for this feature.

Comment (daavoo) [reply content/docs/command-reference/exp/run.md]: Updated without the short flag for now.

**Pull request merged**